src/dbnode/storage/stats/query_stats_default_tracker.go

```diff
@@ -53,27 +60,50 @@ func DefaultQueryStatsTracker(
 		MetricsScope().
 		SubScope("query-stats")
 	return &queryStatsTracker{
-		options:    queryStatsOpts,
-		recentDocs: scope.Gauge("recent-docs-per-block"),
-		totalDocs:  scope.Counter("total-docs-per-block"),
+		options:                   queryStatsOpts,
+		recentDocs:                scope.Gauge("recent-docs-matched"),
+		totalDocs:                 scope.Counter("total-docs-matched"),
+		recentBytesRead:           scope.Gauge("recent-bytes-read"),
+		totalBytesRead:            scope.Counter("total-bytes-read"),
+		recentDocsLimitError:      scope.Tagged(map[string]string{"limit": "docs-matched"}).Counter("error"),
+		recentBytesReadLimitError: scope.Tagged(map[string]string{"limit": "bytes-read"}).Counter("error"),
 	}
 }
 
 func (t *queryStatsTracker) TrackStats(values QueryStatsValues) error {
+	// Only update the recent metrics on each reset so
+	// we measure only consistently timed peak values.
+	if values.ResetDocs {
```

Comment:
Do we need to set `ResetDocs` (and `ResetBytesRead`) to false now?

Reply:
We explicitly are setting them in the reset code within `Start`'s reset loop since that is where we actually are performance the setting on either the docs or the bytes. But agreed prob worth explicitly passing false for both of these in the normal `TrackStats` path as well just to be comprehensive.